sources/packages/backend/apps/api/src/route-controllers/restriction/_tests_/e2e/restriction.aest.controller.addInstitutionRestriction.e2e-spec.ts

```diff
@@ -0,0 +1,434 @@
+      institution,
+      program,
+      location,
+      restriction: susRestriction,
+      creator: sharedAuditUser,
+    });
+    const endpoint = `/aest/restriction/institution/${institution.id}`;
+    const token = await getAESTToken(AESTGroups.BusinessAdministrators);
+
+    // Act/Assert
+    await request(app.getHttpServer())
+      .post(endpoint)
+      .send({
+        restrictionId: susRestriction.id,
+        programId: program.id,
+        locationIds,
+        noteDescription: "Add institution restriction note.",
+      })
+      .auth(token, BEARER_AUTH_TYPE)
+      .expect(HttpStatus.UNPROCESSABLE_ENTITY)
+      .expect({
+        message: `The restriction ID ${susRestriction.id} is already assigned and active to the institution, program ID ${program.id}, and at least one of the location ID(s) ${locationIds}.`,
+        errorType: INSTITUTION_RESTRICTION_ALREADY_ACTIVE,
+      });
+  });
+
+  it("Should throw a not found exception when the provided institution is not found.", async () => {
+    // Arrange
+    const endpoint = `/aest/restriction/institution/999999`;
+    const token = await getAESTToken(AESTGroups.BusinessAdministrators);
+
+    // Act/Assert
+    await request(app.getHttpServer())
+      .post(endpoint)
+      .send({
+        restrictionId: 1,
+        programId: 1,
+        locationIds: [1],
+        noteDescription: "Add institution restriction note.",
+      })
+      .auth(token, BEARER_AUTH_TYPE)
+      .expect(HttpStatus.NOT_FOUND)
+      .expect({
+        message: "Institution ID 999999 not found.",
+        error: "Not Found",
+        statusCode: HttpStatus.NOT_FOUND,
+      });
+  });
+
+  it("Should throw an unprocessable entity exception when the location does not exist associated with the institution.", async () => {
+    // Arrange
+    const [institution, program] = await createInstitutionProgramLocations({
+      numberLocationsToCreate: 0,
+    });
+    const endpoint = `/aest/restriction/institution/${institution.id}`;
+    const token = await getAESTToken(AESTGroups.BusinessAdministrators);
+
+    // Act/Assert
+    await request(app.getHttpServer())
+      .post(endpoint)
+      .send({
+        restrictionId: susRestriction.id,
+        programId: program.id,
+        locationIds: [999999],
+        noteDescription: "Add institution restriction note.",
+      })
+      .auth(token, BEARER_AUTH_TYPE)
+      .expect(HttpStatus.UNPROCESSABLE_ENTITY)
+      .expect({
+        message: `At least one of the location ID(s) 999999 were not associated with the institution.`,
+        error: "Unprocessable Entity",
+        statusCode: HttpStatus.UNPROCESSABLE_ENTITY,
+      });
+  });
+
+  it("Should throw an unprocessable entity exception when the program does not exist associated with the institution.", async () => {
+    // Arrange
+    const [institution, , locationIds] =
+      await createInstitutionProgramLocations({
+        skipProgramCreation: true,
```

Comment:
Minor again. Is there a benefit of skipping program creation from the factory perspective.?

Reply:
First, it avoids saving the program when it is not required for a particular scenario.
Second, it makes explicit in the consumer data setup that a program is not being created.
Besides that, this method is an E2E helper to reduce duplication during data setup. Calling it a factory may be misleading.